tests/config/dapr_crypto_azurekeyvault.yaml

```diff
@@ -0,0 +1,24 @@
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: azurekeyvault
+spec:
+  type: crypto.azure.keyvault
+  version: v1
+  metadata:
+    - name: vaultName
+      envRef: AzureKeyVaultName
+    - name: azureTenantId
+      secretKeyRef:
+        name: azurekeyvault-secret
+        key: tenant-id
+    - name: azureClientId
+      secretKeyRef:
+        name: azurekeyvault-secret
+        key: client-id
+    - name: azureClientSecret
+      secretKeyRef:
+        name: azurekeyvault-secret
+        key: client-secret
```

Comment:
Does this component require actual Azure creds? If so can a specific one be created for the test alone? And not the general Azure Creds for the test also.

Reply:
The problem is that we need a service principal. We could create a separate one, but then we need to store those credentials somewhere and we need to rotate them.

Maybe I can look into using MI or workload identity. Which would be nice to have too. That would allow us to avoid the need for the client secret?

Reply:
As discussed on discord, it seems that neither MI nor WI are convenient in a situation like ours, where clusters are ephemeral, as they require significant setup.

The alternative is to create a separate Azure AD application, however that comes with complexities too as it would be another set of client credentials, which need to be stored as secrets in the repo by an admin and need to be rotated every 6 months too. The complexity is not small either.

As mentioned on Discord, I am not convinced using the same credentials in this specific case is necessarily a problem. While what you say is indeed a best practice for production, in this case I am not sure there is any increased risk in doing what this PR does. The credentials are only available to GitHub Actions that are executed through "/ok-to-test", so only available to trusted users. There's also no _additional_ risk of exfiltration besides what exists today (a malicious actor could modify the pipeline or the `setup_azure.sh` script to make a cURL request to steal the token). Lastly, it should be noted that the subscription associated contains only resources used for tests and no other data.

Reply:
I am fine sharing the creds here. The creds are accessible in the workflow no matter what. What protects the credentials is the requirement to review the PR before sensing /ok-to-test. User can access the secrets from github during workflow after that no matter what.